restli-tools/src/main/java/com/linkedin/restli/tools/snapshot/gen/PegasusSchemaSnapshotExporter.java

```diff
@@ -0,0 +1,162 @@
+   * @param outputDir output files directory
+   * @return GeneratorResult
+   * @throws IOException
+   */
+  public GeneratorResult export(String resolverPath, String inputPath, File outputDir) throws IOException
+  {
+    List<DataSchema> dataSchemas = parseDataSchema(resolverPath, inputPath);
+    pegasusSchemaSnapshotResult result = new pegasusSchemaSnapshotResult();
+    for (DataSchema dataSchema : dataSchemas)
+    {
+     File file = writeSnapshotFile(outputDir, ((NamedDataSchema) dataSchema).getFullName(), dataSchema);
+     result.addModifiedFile(file);
+     result.addTargetFile(file);
+    }
+    return result;
+  }
+
+  private static List<DataSchema> parseDataSchema(String resolverPath, String inputPath)
+      throws RuntimeException, FileNotFoundException
+  {
+    Iterator<File> iterator = FileUtils.iterateFiles(new File(inputPath), new String[]{PDL}, true);
+    List<DataSchema> schemas = new ArrayList<>();
+    while (iterator.hasNext())
+    {
+      DataSchemaResolver resolver = MultiFormatDataSchemaResolver.withBuiltinFormats(resolverPath);
+      File inputFile = iterator.next();
+      PdlSchemaParser parser = new PdlSchemaParser(resolver);
+      parser.parse(new FileInputStream(inputFile));
+
+      if (parser.hasError())
+      {
+        throw new RuntimeException(parser.errorMessage());
+      }
+
+      List<DataSchema> topLevelDataSchemas = parser.topLevelDataSchemas();
+      if (topLevelDataSchemas == null || topLevelDataSchemas.isEmpty() || topLevelDataSchemas.size() > 1)
+      {
+        throw new RuntimeException("Could not parse schema : " + inputFile.getAbsolutePath());
+      }
+      DataSchema topLevelDataSchema = topLevelDataSchemas.get(0);
+      if (!(topLevelDataSchema instanceof NamedDataSchema))
+      {
+        throw new RuntimeException("Invalid schema : " + inputFile.getAbsolutePath() + ", the schema is not a named schema.");
+      }
+      schemas.add(topLevelDataSchema);
+    }
+    return schemas;
+  }
+
+  private static File writeSnapshotFile(File outputDir, String fileName, DataSchema dataSchema) throws IOException
+  {
+    StringWriter stringWriter = new StringWriter();
+    SchemaToPdlEncoder schemaToPdlEncoder = new SchemaToPdlEncoder(stringWriter);
+    schemaToPdlEncoder.setTypeReferenceFormat(AbstractSchemaEncoder.TypeReferenceFormat.DENORMALIZE);
+    schemaToPdlEncoder.encode(dataSchema);
+
+    File generatedSnapshotFile = new File(outputDir, fileName + PEGASUS_SCHEMA_SNAPSHOT_SUFFIX);
+
+    if (generatedSnapshotFile.exists())
+    {
+      if (!generatedSnapshotFile.delete())
+      {
+        throw new IOException(generatedSnapshotFile + ": Can't delete previous version");
+      }
+    }
+
+    try (FileOutputStream out = new FileOutputStream(generatedSnapshotFile))
+    {
+      out.write(stringWriter.toString().getBytes());
+    }
+    return generatedSnapshotFile;
+  }
+
+  private static class pegasusSchemaSnapshotResult implements GeneratorResult
```

Comment:
What is the purpose of this result? Please add documentation

Reply:
I added it for testing purpose.

Reply:
Since it is not used anywhere, removed.